src/messages.rs

```diff
@@ -364,6 +365,98 @@ pub enum ResponseContent {
     },
 }
 
+impl fmt::Debug for DirectMessage {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        match *self {
+            DirectMessage::BootstrapIdentify { ref public_id, ref current_quorum_size } =>
+                write!(formatter, "DirectMessage {{ BootstrapIdentify {{ {:?}, {:?} }} }}",
+                    public_id, current_quorum_size),
+            DirectMessage::BootstrapDeny =>
+                write!(formatter, "DirectMessage {{ BootstrapDeny }}"),
+            DirectMessage::ClientIdentify { ref client_restriction, .. } =>
+                write!(formatter, "DirectMessage {{ ClientIdentify {{ .., {:?} }} }}", client_restriction),
+            DirectMessage::NodeIdentify { .. } =>
+                write!(formatter, "DirectMessage {{ NodeIdentify {{ .. }} }}"),
```

Comment:
Maybe the `serialised_public_id` should be included for `ClientIdentify` and `NodeIdentify`, using `format_binary_array`.